libraries/psio/include/psio/reflect.hpp

```diff
@@ -0,0 +1,760 @@
+   {
+      return {a, FixedString{b}};
+   }
+
+   struct meta
+   {
+      const char*                        name;
+      uint64_t                           offset      = ~uint64_t(0);
+      std::initializer_list<const char*> param_names = {};
+      int32_t                            number      = 0;
+   };
+
+   template <typename QueryClass>
+   struct reflect_undefined
+   {
+      static constexpr bool is_defined              = false;
+      static constexpr bool is_struct               = false;
+      static constexpr bool definitionWillNotChange = false;
+      template <typename L>
+      static void get(const std::string_view& m, L&& lambda);
+   };
+
+   template <typename QueryClass>
+   reflect_undefined<QueryClass> get_reflect_impl(const QueryClass&);
+
+   template <typename QueryClass>
+   using reflect = std::decay_t<decltype(get_reflect_impl(std::declval<QueryClass>()))>;
+
+   template <typename>
+   struct is_std_vector : std::false_type
+   {
+   };
+
+   template <typename T, typename A>
+   struct is_std_vector<std::vector<T, A>> : std::true_type
+   {
+      using value_type = T;
+   };
+
+   template <typename T>
+   constexpr bool is_std_vector_v = is_std_vector<T>::value;
+
+   template <typename>
+   struct is_std_optional : std::false_type
+   {
+   };
+
+   template <typename T>
+   struct is_std_optional<std::optional<T>> : std::true_type
+   {
+      using value_type = T;
+   };
+
+   template <typename T>
+   constexpr bool is_std_optional_v = is_std_optional<T>::value;
+
+   template <typename>
+   struct is_std_unique_ptr : std::false_type
+   {
+   };
+
+   template <typename T>
+   struct is_std_unique_ptr<std::unique_ptr<T>> : std::true_type
+   {
+      using value_type = T;
+   };
+
+   template <typename T>
+   constexpr bool is_std_unique_ptr_v = is_std_unique_ptr<T>::value;
+
+   template <typename>
+   struct is_std_reference_wrapper : std::false_type
+   {
+   };
+
+   template <typename T>
+   struct is_std_reference_wrapper<std::reference_wrapper<T>> : std::true_type
+   {
+      using value_type = T;
+   };
+
+   template <typename T>
+   constexpr bool is_std_reference_wrapper_v = is_std_reference_wrapper<T>::value;
+
+   template <typename>
+   struct is_std_array : std::false_type
+   {
+   };
+
+   template <typename T, auto N>
+   struct is_std_array<std::array<T, N>> : std::true_type
+   {
+      using value_type                        = T;
+      constexpr static const std::size_t size = N;
+   };
+
+   template <typename T>
+   constexpr bool is_std_array_v = is_std_array<T>::value;
+
+   template <typename>
+   struct is_std_variant : std::false_type
+   {
+   };
+
+   template <typename... T>
+   struct is_std_variant<std::variant<T...>> : std::true_type
+   {
+      static std::string name() { return get_variant_typename<T...>(); }
+      template <typename First, typename... Rest>
+      static std::string get_variant_typename()
```

Comment:
This is not used, and the name is also not unique. For example, what type is int&double|char? This is also using `get_type_name` to get the representation of the members, which does something totally different (and equally wrong). We need to consider where the names are being used (and therefore what characters are permitted) and work out a single encoding.